Assets/HoloToolkit-Examples/Utilities/Scripts/SwapVolume.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule.Tests
+{
+    /// <summary>
+    /// This class is used in the SolverExamples scene, used to swap between active solvers
+    /// and placeholder solvers displayed in the scene.
+    /// </summary>
+    public class SwapVolume : MonoBehaviour, IInputClickHandler
+    {
+        [SerializeField]
+        private GameObject hideThisObject = null;
+
+        [SerializeField]
+        private GameObject spawnThisPrefab = null;
+
+        [SerializeField]
+        private bool updateSolverTargetToClickSource = true;
+
+        private SolverHandler solverHandler;
+        private bool isOn = false;
+        private GameObject spawnedObject;
+
+        private void Start()
+        {
+            spawnedObject = Instantiate(spawnThisPrefab, hideThisObject.transform.position, hideThisObject.transform.rotation);
+            spawnedObject.SetActive(false);
+            solverHandler = spawnedObject.GetComponent<SolverHandler>();
```

Comment:
what should happen if the object does not have the SolverHandler component attached? should at least log a warning

Reply:
Since this script is only used in the solver example scene, the `spawnedObject` is guaranteed to be set up correctly.